error[E0521]: borrowed data escapes outside of associated function
  --> $DIR/issue-72312.rs:13:24
   |
LL |       pub async fn start(&self) {
   |                          ----- `self` is a reference that is only valid in the associated function body
...
LL |           require_static(async move {
   |  ________________________^
LL | |             &self;
LL | |         });
   | |_________^ `self` escapes the associated function body here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0521`.
